yaml_cpp_vendor-extras.cmake.in

```diff
@@ -5,7 +5,7 @@ if(NOT yaml-cpp_FOUND)
   if(WIN32)
     set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/CMake")
   else()
-    set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/lib/cmake/yaml-cpp")
+    set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/share/cmake/yaml-cpp")
```

Comment:
The commit message for this change is rather vague. I'm not sure why the change was made.

Reply:
I cannot reproduce this yaml-cpp failure locally on macOS Monterey.  I tried both with and without `yaml-cpp` installed via brew, and in both cases all of the tests succeeded for me.  At this point, I can only assume that the failure was happening because of some interaction on the old macOS Mojave builders, but since we are no longer supporting that I don't think this should be held up on that.  I'm going to go ahead and make the changes as requested here, and then turn this into a full PR and run CI on it again.